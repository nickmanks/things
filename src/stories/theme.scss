
$storybook-container-shadow: 1px 1px 5px 1px rgba(0, 0, 0, .37);

.story-container {
  margin: 3rem;
  box-shadow: $storybook-container-shadow;
  width: 100%;
  height: 65rem;
  border-radius: .4rem;

  > div {
    height: 100%;
    width: 100%;

    > div:first-child {
      height: 100%;
      width: 100%;
    }
  }
}
