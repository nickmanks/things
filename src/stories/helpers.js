import React from 'react';
import {createStore, applyMiddleware} from 'redux';
import thunk from 'redux-thunk';
import reducers from '../reducers';
import 'bootstrap/dist/css/bootstrap.min.css';
import 'shards-ui/dist/css/shards.min.css';
import './theme.scss';


export function withCustomContainer(backgroundColor) {
  const CustomContainer = (story)=> (
    <div className={'story-container'} style={{backgroundColor}}>
      {story()}
    </div>
  );
  CustomContainer.displayName = 'CustomContainer';
  return CustomContainer;
}

export const storyStore =
  (state)=> createStore(reducers, state, applyMiddleware(thunk));
