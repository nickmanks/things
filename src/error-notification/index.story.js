/* global module */
import React from 'react';
import {storiesOf} from '@storybook/react';
import {storyStore} from '../stories/helpers';
import ErrorNotification from '.';


storiesOf('Connected/ErrorNotification', module)
  .add('with no errors', ()=> {
    const store = storyStore({
      persistence: {
        loaded: true,
        queued: [],
        processing: [],
        done: [],
        errors: []
      }
    });

    return (<ErrorNotification store={store} />);
  })
  .add('with errors', ()=> {
    const store = storyStore({
      persistence: {
        loaded: true,
        queued: [],
        processing: [],
        done: [],
        errors: ['error']
      }
    });

    return (<ErrorNotification store={store} />);
  });
