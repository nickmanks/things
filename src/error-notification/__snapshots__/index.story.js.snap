// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Connected/ErrorNotification with errors 1`] = `
<ErrorNotification store={{}} message={"Unable to connect at this time."} removeError={"function"} storeSubscription={{"store":{},"listeners":{}}}>
  <Fade in={true} className={"fade-in-error"} mountOnEnter={false} unmountOnExit={false} appear={true} enter={true} exit={true} onEnter={"function"} onEntering={"function"} onEntered={"function"} onExit={"function"} onExiting={"function"} onExited={"function"} tag={"div"} baseClass={"fade"} baseClassActive={"show"} timeout={150}>
    <Transition in={true} mountOnEnter={false} unmountOnExit={false} appear={true} enter={true} exit={true} timeout={150} onEnter={"function"} onEntering={"function"} onEntered={"function"} onExit={"function"} onExiting={"function"} onExited={"function"} />
  </Fade>
</ErrorNotification>
`;

exports[`Storyshots Connected/ErrorNotification with no errors 1`] = `
<ErrorNotification store={{}} message={null} removeError={"function"} storeSubscription={{"store":{},"listeners":{}}}>
  <Fade in={false} className={"fade-in-error"} mountOnEnter={false} unmountOnExit={false} appear={true} enter={true} exit={true} onEnter={"function"} onEntering={"function"} onEntered={"function"} onExit={"function"} onExiting={"function"} onExited={"function"} tag={"div"} baseClass={"fade"} baseClassActive={"show"} timeout={150}>
    <Transition in={false} mountOnEnter={false} unmountOnExit={false} appear={true} enter={true} exit={true} timeout={150} onEnter={"function"} onEntering={"function"} onEntered={"function"} onExit={"function"} onExiting={"function"} onExited={"function"} />
  </Fade>
</ErrorNotification>
`;
